lib/grpc/rpc_error.ex

```diff
@@ -71,7 +71,7 @@ defmodule GRPC.RPCError do
     %{error | message: error.message || Status.status_message(error.status)}
   end
 
-  def from_status(401), do: exception(GRPC.Status.unauthorized(), status_message(401))
+  def from_status(401), do: exception(GRPC.Status.unauthenticated(), status_message(401))
```

Comment:
401 is unauthorized as per HTTP specs

Reply:
https://developer.mozilla.org/pt-BR/docs/Web/HTTP/Status/401

Reply:
I believe we should use the same verbs as HTTP if we are to do any kind of translation here

Reply:
Hi, @polvalente. There's only so much we can do without adding new errors to the library. As the GRPC specification states only 16 error codes, what we do is only a best efforth. I copied all the matching pairs from the official implementation in Go

Reply:
Right! In this case, I believe a comment should be added noting this :)

Reply:
WRT to the mint implementation, I can add, but I wanted to get some validation on this first :)

Reply:
Nice :)
Go ahead with Mint and tests then! Looks good to me now, with the clarification above

Reply:
Hi, @polvalente. I managed to write tests, but found that adding the status handling to the mint adapter is way more complex due to the way it's structured. Do they really need to be feature-equivalent? Can't we perhaps open an issue for a mint implementation of this feature in the future?

Reply:
I'd rather have them being feature-equivalent as one is meant to be a drop-in replacement of the other.

Can you elaborate on what made it difficult to add this kind of handling?

Reply:
Essentially, the status is received at `ConnectionProcess`, which is called to start the request, but the calling process reads the response from the `StreamResponseProcess`. Meaning I'd need to write some process coordination or rework current coordination logic. Since I don't fully understand the current architecture decisions (yet?), I found this a bit challenging.

Reply:
I believe you need to change things here https://github.com/elixir-grpc/grpc/blob/b3a7ff627ea013907a78c693016325a023a540b8/lib/grpc/client/adapters/mint.ex#L162C3-L164C6

Reply:
And maybe here: https://github.com/elixir-grpc/grpc/blob/b3a7ff627ea013907a78c693016325a023a540b8/lib/grpc/client/adapters/mint.ex#L162C3-L164C6